cg/meta/upload/nipt/nipt.py

```diff
@@ -51,18 +49,32 @@ def set_dry_run(self, dry_run: bool) -> None:
         self.dry_run = dry_run
 
     def flowcell_passed_qc_value(self, case_id: str, q30_threshold: float) -> bool:
-        """Check average Q30 and of the latest flow cell related to a case."""
-        latest_flow_cell: Flowcell = self.status_db.get_latest_flow_cell_on_case(family_id=case_id)
-        flow_cell_reads_and_q30_summary: Dict[
-            str, Union[int, float]
-        ] = self.stats_api.flow_cell_reads_and_q30_summary(flow_cell_name=latest_flow_cell.name)
-        return flow_cell_reads_and_q30_summary[
-            "q30"
-        ] >= q30_threshold and flow_cell_reads_and_q30_summary[
-            "reads"
-        ] >= self.status_db.get_ready_made_library_expected_reads(
-            case_id=case_id
+        """
+        Check the average Q30 and the total number of reads for each sample
+        in the latest flow cell related to a case.
+        """
+        flow_cell: Flowcell = self.status_db.get_latest_flow_cell_on_case(family_id=case_id)
+        flow_cell_summary: FlowCellQ30AndReads = FlowCellQ30AndReads(
+            average_q30_across_samples=self.status_db.get_average_fraction_passing_q30_from_sample_lane_metrics(
```

Comment:
A matter of taste, but I don't think the caller needs to know which table in the database the information is retrieved from. I would say that is an implementation detail of the function and it is not something we want the caller to think about. The caller is only interested in that the function returns the number of reads associated with a flow cell, how it is done is not of interest to the caller and is just an implementation detail.

So (in my opinion) names like these would be better because they don't leak implementation details
```python
get_average_fraction_passing_q30_for_flow_cell(flow_cell.name)
get_number_of_reads_for_flow_cell(flow_cell.name)
```

Reply:
I shortened the function names!